Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -145,6 +145,12 @@ protected override async void Start()
             }
 
             SetCursor();
+
+            BaseCursor c = BaseCursor as BaseCursor;
```

Comment:
Another thing to consider - it's possible for BaseCursor to change as a result of calling SetCursor - if VisibleSourcesCount has been changed on BaseCursor, then we'd still be off in some way (i..e if we changed it, VisibleSourcesCount would be up by one on the old BaseCursor, and wouldn't have been correctly incremented on the new one)

Reply:
In what situations are SetCursor called?

Reply:
I have noooooo idea, just mentioning this after looking at the code

Reply:
I have a special place in my heart for refcounts.